src/runtime/internal/utils.ts

```diff
@@ -105,6 +105,13 @@ export function exclude_internal_props(props) {
 	return result;
 }
 
+export function compute_rest_props(props, keys) {
+	const rest = {};
+	keys = new Set(keys);
+	for (const k in props) if (!keys.has(k) && k[0] !== '$') rest[k] = props[k];
+	return Object.freeze(rest);
```

Comment:
Is there a particular reason to freeze this when `$$props` isn't frozen?

Reply:
I just think that `$$rest` shouldn't be mutated, because it's kind of derived from `$$props`, and mutation on `$$rest` will not propagate back to the `$$props`

Reply:
It doesn't have to be ironed out before this can get merged, but what are your thoughts on allowing `$$props` to be mutated? I think that right now it can but those changes won't be reflected in the corresponding other variables.

Reply:
Actually, initially I was thinking of only freezing in dev mode, to educate users not to mutate it.

Not sure Object.freeze has any performance penalties... 

What do u think?

Reply:
On another thought, I remove the freeze for now, to be consistent with the $$props.

In a separate PR, I'll add warnings for assignment or mutation on $$props and $$rest